scarb-metadata/src/version_pin.rs

```diff
@@ -49,9 +49,23 @@ impl Serialize for VersionPin {
 impl<'de> Deserialize<'de> for VersionPin {
     fn deserialize<D: Deserializer<'de>>(d: D) -> Result<VersionPin, D::Error> {
         use serde::de::Error;
-        let num = u64::deserialize(d)?;
-        VersionPin::from_numeric(num)
-            .ok_or_else(|| Error::custom(format!("expected metadata version {}", Self.numeric())))
+        match u64::deserialize(d) {
+            Ok(num) => VersionPin::from_numeric(num).ok_or_else(|| {
+                Error::custom(format!("expected metadata version {}", Self.numeric()))
+            }),
+            Err(err) => Err({
+                let err_message = err.to_string();
+
+                if err_message.contains("missing field") {
+                    err
+                } else {
+                    Error::custom(format!(
+                        "{}\n expected metadata version to be of type u64",
+                        err_message
+                    ))
+                }
```

Comment:
Why does it need to change?

Reply:
So it can be captured in `parse_stream` for returned error consistency

Reply:
Yes, I see that you capture it later, but why can't we simply return `MetadataCommandError::Json` regardless of the underlying deserialization error source?

Reply:
I implemented it that way for compatibility (see tests). I agree it would be simpler and should not matter. If you are sure it won't break anything i can change it.